### PR TITLE
Added methods to find the version and tranche configuration

### LINKED ISSUE #1: Print current version
Print current software version reading from Cargo.toml

### LINKED ISSUE #2: Read wallet and cluster parameters from solana config

### LINKED ISSUE #5: Introduce cli common libraries
Evaluate the followings:

    https://github.com/console-rs/console
    https://docs.rs/indicatif/latest/indicatif/
    https://docs.rs/dialoguer/latest/dialoguer/

Check dependencies used [here](https://github.com/solana-labs/solana/blob/v1.10.39/cli/Cargo.toml)

### LINKED ISSUE #6: Read Vyper Core Tranche Configuration

### PR DESCRIPTION
**Pull request by iamakshat01**
1. Implemented vyper --version. Closes #1.
2. Implemented vyper core fetch trancheId. Closes #6 
3. Used solana-cli-config to configuration. Closes #2 
4. Introduced console and clap. Closes #5 #4 
5. Working on having a progress bar while fetching tranche configuration.

### CODE REVIEW

File: src/ops/core_ops.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            println_name_value("junior tranche_mint", &account.junior_tranche_mint);
+            println_name_value("tranche authority", &account.tranche_authority);
+            println_name_value("authority seed", &account.authority_seed);
+            println_name_value("authority bump", &account.authority_bump);
+            println_name_value("owner", &account.owner);
+            println_name_value("rate program", &account.rate_program);
+            println_name_value("rate program state", &account.rate_program_state);
+            println_name_value("redeem logic program", &account.redeem_logic_program);
+            println_name_value("redeem logic program state", &account.redeem_logic_program_state);
+            println_name_value("version", &account.version);
```
Comment (giacomo-cc) [inline src/ops/core_ops.rs]: can we format from this `version : [0, 1, 0]` to this `version: 0.1.0`

File: src/ops/core_ops.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+pub fn handle_core_command(core_command: CoreCommand, program: &Program) {
+    let command = core_command.command;
+    match command {
+        CoreSubcommand::Fetch(fetch_tranche) => {
+            let account: TrancheConfig = program.account(fetch_tranche.tranche_id).expect("Could not find tranche with given publickey");
+            println_name_value("reserve mint",&account.reserve_mint);
+            println_name_value("reserve",&account.reserve);
+            println_name_value("deposited quantity", &account.tranche_data.deposited_quantity);
+            println_name_value("reserve fair value", &account.tranche_data.reserve_fair_value);
+            println_name_value("tranche fair value", &account.tranche_data.reserve_fair_value);
```
Comment (giacomo-cc) [inline src/ops/core_ops.rs]: here probably we have a wrong copy and paste, should be:
```rust
println_name_value("reserve fair value", &account.tranche_data.reserve_fair_value);
println_name_value("tranche fair value", &account.tranche_data.tranche_fair_value);
```

Comment (giacomo-cc) [inline src/ops/core_ops.rs]: for tranche and reserve fair values we're using under the hood [this library](https://docs.rs/rust_decimal/latest/rust_decimal/), serializing bytes on chain for anchor-compatibility (further reading [here](https://twitter.com/vanderlinde____/status/1564308344759656448?s=20&t=3P2qGBQt8KK_CuIEx0f5nw)). 

Can we try to deserialize decimal values with [Decimal::deserialize](https://docs.rs/rust_decimal/latest/rust_decimal/prelude/struct.Decimal.html#method.deserialize) transforming this:
```
reserve fair value : ReserveFairValue { value: [[0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], slot_tracking: SlotTracking { last_update: LastUpdate { slot: 163978618, _padding: [0, 0, 0, 0, 0, 0, 0, 0] }, stale_slot_threshold: 2 } } 
```
to this
```
reserve fair value : ReserveFairValue { value: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1], slot_tracking: SlotTracking { last_update: LastUpdate { slot: 163978618, _padding: [0, 0, 0, 0, 0, 0, 0, 0] }, stale_slot_threshold: 2 } } 
```

**Review by giacomo-cc — CHANGES_REQUESTED** [review]

File: src/ops/core_ops.rs (hunk 2 above)
Comment (iamakshat01) [reply src/ops/core_ops.rs]: Ok

Comment (iamakshat01) [reply src/ops/core_ops.rs]: Fixing

File: src/ops/core_ops.rs (hunk 1 above)
Comment (iamakshat01) [reply src/ops/core_ops.rs]: Ok

**Review by giacomo-cc — APPROVED** [review]
looks good to me

**Pull request merged**